cmd/clusterctl/client/rollout.go

```diff
@@ -69,6 +69,70 @@ func (c *clusterctlClient) RolloutRestart(options RolloutRestartOptions) error {
 	return nil
 }
 
+func (c *clusterctlClient) RolloutPause(options RolloutOptions) error {
+	clusterClient, err := c.clusterClientFactory(ClusterClientFactoryInput{Kubeconfig: options.Kubeconfig})
+	if err != nil {
+		return err
+	}
+
+	// If the option specifying the Namespace is empty, try to detect it.
+	if options.Namespace == "" {
+		currentNamespace, err := clusterClient.Proxy().CurrentNamespace()
+		if err != nil {
+			return err
+		}
+		options.Namespace = currentNamespace
+	}
+
+	if len(options.Resources) == 0 {
+		return fmt.Errorf("required resource not specified")
+	}
+	normalized := normalizeResources(options.Resources)
+	tuples, err := util.ResourceTypeAndNameArgs(normalized...)
+	if err != nil {
+		return err
+	}
```

Comment:
This code seems duplicated in every method. I suggest refactoring it to consolidate this.

Reply:
Refactored it.